src/gkeClient.ts

```diff
@@ -60,8 +60,8 @@ export class ClusterClient {
     let projectId = opts?.projectId;
     if (
       !opts?.credentials &&
-      (!process.env.GCLOUD_PROJECT ||
```

Comment:
We can get rid of this and rely on the auth lib so we dont have to maintain it.

Reply:
Sorry I'm not sure I follow, do we still want to throw when `opts.credentials` is null and `process.env.GCLOUD_PROJECT` is manually provided by the workflow?
If I'm leveraging the runner's Application Default Credentials do I still need to use the auth lib?